### PR TITLE
chore: send alerts can wait forever and fail for broken workflows

### PR DESCRIPTION
**Pull request by hamidzr**
## Ticket

https://hpe-aiatscale.atlassian.net/browse/RM-372

## Description

adding a condition to skip over workflows that have elapsed their timeout instead of reporting them as active.
make sure send alerts does not:

### why

send-alert job:
1. takes too long 5hrs in these cases
2. does not fail if another ci job takes long

this job has been staying active for 5hours
https://app.circleci.com/pipelines/github/determined-ai/determined/57843/workflows/1a75f6bf-77b2-45e5-8155-9f2f5dcc96c4/jobs/2742238

testing the timeout https://app.circleci.com/pipelines/github/determined-ai/determined/58023/workflows/53afd482-abd0-403a-9346-0de85400e533/jobs/2756232

context https://hpe-aiatscale.slack.com/archives/C04C9JXB1C2/p1720448458815989

## Checklist

- [x] Changes have been manually QA'd
- [ ] New features have been approved by the corresponding PM
- [ ] User-facing API changes have the "User-facing API Change" label
- [ ] Release notes have been added as a separate file under `docs/release-notes/`
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses have been included for new code which was copied and/or modified from any external code

[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .circleci/scripts/job_level_alerts.py [hunk 1]
```diff
@@ -32,18 +33,27 @@ def send_alert(job_name: str, pipeline_number: str, workflow_id: str, job_number
...
+
     for w in workflows["items"]:
         if w["name"] in workflows_to_skip:
             continue
 
         workflow_id = w["id"]
         if not workflows_are_running and w["stopped_at"] is None:
-            print(f"waiting for at least workflow {w['name']} to finish")
-            workflows_are_running = True
+            created_at = datetime.datetime.strptime(w["created_at"], "%Y-%m-%dT%H:%M:%SZ")
```
Comment (JComins000) [inline .circleci/scripts/job_level_alerts.py:50]: how about this
```python
            created_at = datetime.datetime.strptime(w["created_at"], "%Y-%m-%dT%H:%M:%SZ")
            if created_at < earliest_accepted_time:
                print(f"workflow {w['name']} timed out.")
                # TODO: add support for reporting as a timeout or failure.
                continue
            print(f"waiting for at least workflow {w['name']} to finish")
            workflows_are_running = True
```

**Comment by JComins000** [comment]
why don't you make an issue for this? the issue content should explain the problem or "why" you're doing this work. it's your current pr description.
the pr description should explain the implementation or "how" you're doing the work. something like "adding a condition to skip over workflows that have elapsed their timeout instead of always setting `workflowsRunning` to `true`

File: .circleci/scripts/job_level_alerts.py (hunk 1 above)
Comment (JComins000) [reply .circleci/scripts/job_level_alerts.py:50]: sure, it's just style at the end of the day

**Review by JComins000 — APPROVED** [review]

**Comment by hamidzr** [comment]
thanks for the review!

**Pull request merged**